reactive/src/test/java/feign/reactive/ReactiveInvocationHandlerTest.java

```diff
@@ -62,17 +64,33 @@ public void invokeOnSubscribeReactor() throws Throwable {
     verifyZeroInteractions(this.methodHandler);
 
     /* subscribe and execute the method */
-    Mono mono = (Mono) result;
-    mono.log().block();
+    StepVerifier.create((Mono) result)
+            .expectNext("Result")
+            .expectComplete()
+            .verify();
+    verify(this.methodHandler, times(1)).invoke(any());
+  }
+
+  @Test
+  public void invokeOnSubscribeEmptyReactor() throws Throwable {
+    given(this.methodHandler.invoke(any())).willReturn(null);
+    ReactorInvocationHandler handler = new ReactorInvocationHandler(this.target,
+            Collections.singletonMap(method, this.methodHandler));
+
+    Object result = handler.invoke(method, this.methodHandler, new Object[] {});
+    assertThat(result).isInstanceOf(Mono.class);
+    verifyZeroInteractions(this.methodHandler);
+
+    /* subscribe and execute the method */
+    StepVerifier.create((Mono) result)
+            .expectComplete()
+            .verify();
     verify(this.methodHandler, times(1)).invoke(any());
   }
 
-  @SuppressWarnings("unchecked")
   @Test
   public void invokeFailureReactor() throws Throwable {
-    this.thrown.expect(RuntimeException.class);
-    given(this.methodHandler.invoke(any())).willThrow(new RuntimeException("Could Not Decode"));
-    given(this.method.getReturnType()).willReturn((Class) Class.forName(Mono.class.getName()));
```

Comment:
This test didn't even run because it throws NullPointerException because this.method was never set. And the test passes because NullPointerException is runtime exception and ExpectedException was marking the test as passed.

Reply:
Thank you for pointing that out.